datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -262,6 +268,53 @@ fn date_trunc_single(granularity: &str, value: i64) -> Result<i64> {
     Ok(value.unwrap().timestamp_nanos())
 }
 
+fn _date_trunc(
+    tu: TimeUnit,
+    value: &Option<i64>,
+    granularity: &str,
+    f: impl Fn(Option<i64>) -> Result<Option<i64>>,
+    // tz_opt: &Option<Arc<str>>,
+) -> Result<Option<i64>, DataFusionError> {
+    let scale = match tu {
+        TimeUnit::Second => 1_000_000_000,
+        TimeUnit::Millisecond => 1_000_000,
+        TimeUnit::Microsecond => 1_000,
+        TimeUnit::Nanosecond => 1,
+    };
+
+    if value.is_none() {
+        return Ok(None);
+    }
```

Comment:
This works fine but I think you can simplify the logic with something like the following (and then avoid all the unwraps)

```suggestion
    let Some(value) = value else {
        return Ok(None);
    }
```